How to add hidden pages (#655)
mintlify/docs: guides/hidden-pages.mdx

---

> title: "How to add hidden pages"
> description: "Exclude pages from your docs.json"

Reviewer: ```suggestion
title: "Hidden pages"
description: "Exclude pages from your `docs.json`"
icon: "eye-closed"
```

Titling it "hidden pages" is more similar to how other guides are named and let's add an icon.

---

> Sometimes you may want to write a page and not publish it immediately.
> 
> For content that is not strictly confidential, but isn't ready for prime time, you can add pages to your docs without referencing them in your `docs.json`. This is what is considered a "hidden page".
> 
> As long as the page is not described under any "group", "tab" or other element in the navigation section of your `docs.json`, the only way a user will be able to navigate to the page is if they know or guess the page URL.
> 
> <Note>
> Some navigation elements like sidebars, dropdowns, and tabs may shift or appear empty on hidden pages.
> </Note>

Reviewer: ```suggestion
You can use hidden pages for content that is not strictly confidential, but isn't ready for prime time. Hidden pages are excluded from your site's navigation, so people can only access hidden pages if they know the exact URL.

A page is hidden until you add it to a group, tab, or other navigation element in your `docs.json`.

<Note>
Some navigation elements like sidebars, dropdowns, and tabs may shift or appear empty on hidden pages.
</Note>
```

I love the `For content that is not strictly confidential, but isn't ready for prime time,`. Whimsical and fun, which is how writing docs should be 😆 

Just some suggestions to make this section a bit more concise and to remove the negations from how things work ("is hidden until" instead of "is not described"). In general, using not can trip up users since they might miss **not** and end up thinking the opposite of what's written.

> If you want hidden pages to be indexed for SEO purposes and don't mind if their content appears in search results, you can add the following setting to your `docs.json`:

Reviewer: ```suggestion
If you want hidden pages to be indexed for SEO purposes and for their contents to appear when people search within your docs, add the following setting to your `docs.json`:
```

Combining some info in this section to make it a bit more concise. Had to split into two comments since the code block messes with GitHub's suggestion formatting.

```
"seo" {
    "indexing": all
}
```

> The settings for SEO indexing also influence whether the contents of hidden pages can appear when your docs are searched internally with the search bar. If you want to make sure hidden pages' content does not appear in SEO or internal search results, make sure to add `noindex: true` to the frontmatter of your page.

Reviewer: ```suggestion
If you want to exclude a specific hidden page's content from search engine and internal search results, add `noindex: true` to the frontmatter of the page.
```

> Learn more about SEO and Sitemaps configuration in the [SEO](https://mintlify.com/docs/settings/seo#sitemaps) docs.
> 
> 
> If you have certain pages that you only want exposed to specific groups of your own users, you may want to use the selective groups access available through [Personalization](https://mintlify.com/docs/settings/authentication-personalization/personalization#showing-hiding-pages).
> 
> If you have docs that you want under no circumstances exposed to the wider web, you may want to use [Authentication](https://mintlify.com/docs/settings/authentication-personalization/authentication).

Reviewer: ```suggestion
If you have pages that you want only specific groups of your own users to have access to, you can use the selective groups access available through [Personalization](/settings/authentication-personalization/personalization#showing-hiding-pages).

If you have pages that you want to maintain strict access control to, you can use [Authentication](/settings/authentication-personalization/authentication).
```

Let's cut the first link since if people are interested, they should be able to find it and the second two links are probably more likely to be related to tasks a user is thinking of while reading up on hidden pages. Just a few wording edits and changed the links to relative links.

For content that is not strictly confidential, but isn't ready for prime time, you can add pages to your docs without referencing them in your `docs.json`. This is what is considered a "hidden page".

As long as the page is not described under any "group", "tab" or other element in the navigation section of your `docs.json`, the only way a user will be able to navigate to the page is if they know or guess the page URL.

<Note>
Some navigation elements like sidebars and tabs may shift or appear empty on hidden pages
</Note>

If you want to take further steps to hide hidden pages' content, make sure to add `noindex: true` to the frontmatter of your page. This should prevent the page from appearing in external search results. On the other hand, if you do want hidden pages indexed for SEO purposes you can add the following setting to your `docs.json`:

```
"seo" {
    "indexing": all
}
```

Learn more about SEO and Sitemaps configuration in the [SEO](https://mintlify.com/docs/settings/seo#sitemaps) docs.

<Warning>
Contents of hidden pages will not be hidden from internal docs search, and may appear in AI-generated results

> </Warning>

Author: not sure if this is still true based on Slack conversations

Author: discussed with Sam and Mayank, the SEO setting applies to the internal search/AI search as well. updated in [95bde84](https://github.com/mintlify/docs/pull/655/commits/95bde84be11af568ba1167b9197417f81e45c466)
